src/main/kotlin/vm/Executable.kt

```diff
@@ -0,0 +1,35 @@
+package com.dlfsystems.vm
+
+import com.dlfsystems.value.VFun
+import com.dlfsystems.value.VObj
+import com.dlfsystems.value.Value
+import kotlinx.serialization.Serializable
+
+interface Executable {
+
+    @Serializable
+    data class Block(val start: Int, val end: Int)
+
+    val code: List<VMWord>
+    val symbols: Map<String, Int>
+    val blocks: List<Block>
+
+    fun getLambda(
+        block: Int,
+        vThis: VObj,
+        args: List<String>,
+        withVars: Map<String, Value>,
+    ): VFun {
+        val offset = blocks[block].start
+        val code = code.subList(blocks[block].start, blocks[block].end).map { word ->
+            // Rewrite jump dests with offset
+            word.address?.let { VMWord(
+                lineNum = word.lineNum,
+                charNum = word.charNum,
+                address = it - offset
+            ) } ?: word
+        }
+        return VFun(code, symbols, blocks, vThis, args, withVars)
```

Comment:
We copy the full symbol and block tables from the parent executable; this is wasteful but simple.  Only things actually referenced in the lambda will matter; we could select those out but I didn't.